torchaudio/functional.py

```diff
@@ -1364,18 +1365,18 @@ def _generate_wave_table(
         min (float): desired min value
         max (float): desired max value
         phase (float): desired phase
-
+        device (torch.device): Torch device on which table must be generated
     Returns:
         Tensor: A 1D tensor with wave table values
     """
 
     phase_offset = int(phase / math.pi / 2 * table_size + 0.5)
 
-    t = torch.arange(table_size).to(torch.int32)
+    t = torch.arange(table_size, device=device).to(torch.int32)
```

Comment:
You can directly path both `drive` and `dtype` arguments to `torch.arange`.

https://pytorch.org/docs/master/generated/torch.arange.html#torch-arange